training_step/resnet/train.py

```diff
@@ -0,0 +1,330 @@
+        Initializes the optimizer and learning rate scheduler
+
+        :return: output - Initialized optimizer and scheduler
+        """
+        self.optimizer = torch.optim.Adam(self.parameters(), lr=self.args["lr"])
+        self.scheduler = {
+            "scheduler": torch.optim.lr_scheduler.ReduceLROnPlateau(
+                self.optimizer,
+                mode="min",
+                factor=0.2,
+                patience=3,
+                min_lr=1e-6,
+                verbose=True,
+            ),
+            "monitor": "val_loss",
+        }
+        return [self.optimizer], [self.scheduler]
+
+
+def train_model(
+    train_glob: str,
+    gpus: int,
+    tensorboard_root: str,
+    max_epochs: int,
+    train_batch_size: int,
+    val_batch_size: int,
+    train_num_workers: int,
+    val_num_workers: int,
+    learning_rate: int,
+    accelerator: str,
+    model_save_path: str,
+):
+
+    if accelerator == "None":
+        accelerator = None
+
+    dict_args = {
+        "train_glob": train_glob,
+        "max_epochs": max_epochs,
+        "train_batch_size": train_batch_size,
+        "val_batch_size": val_batch_size,
+        "train_num_workers": train_num_workers,
+        "val_num_workers": val_num_workers,
+        "lr": learning_rate,
+        "accelerator": accelerator,
+    }
+
+    classes = (
+        "plane",
+        "car",
+        "bird",
+        "cat",
+        "deer",
+        "dog",
+        "frog",
+        "horse",
+        "ship",
+        "truck",
+    )
+
+    dm = CIFAR10DataModule(**dict_args)
+    dm.prepare_data()
+    dm.setup(stage="fit")
+
+    model = CIFAR10Classifier(**dict_args)
+    early_stopping = EarlyStopping(monitor="val_loss", mode="min", patience=5, verbose=True)
+
+    Path(model_save_path).mkdir(parents=True, exist_ok=True)
+
+    if len(os.listdir(model_save_path)) > 0:
+        for filename in os.listdir(model_save_path):
+            filepath = os.path.join(model_save_path, filename)
+            try:
+                shutil.rmtree(filepath)
+            except OSError:
+                os.remove(filepath)
+
+    checkpoint_callback = ModelCheckpoint(
+        dirpath=model_save_path,
+        filename="cifar10_{epoch:02d}",
+        save_top_k=1,
+        verbose=True,
+        monitor="val_loss",
+        mode="min",
+        prefix="",
+    )
+
+    Path(tensorboard_root).mkdir(parents=True, exist_ok=True)
+
+    # Tensorboard root name of the logging directory
+    tboard = TensorBoardLogger(tensorboard_root, "cifar10_lightning_kubeflow")
+    lr_logger = LearningRateMonitor()
+
+    trainer = pl.Trainer(
+        gpus=gpus,
+        logger=tboard,
+        checkpoint_callback=checkpoint_callback,
+        max_epochs=max_epochs,
+        callbacks=[lr_logger, early_stopping],
+        accelerator=accelerator
+    )
+
+    trainer.fit(model, dm)
+    trainer.test()
+
+    return checkpoint_callback.best_model_path
+
+
+# if __name__ == "__main__":
+#     train_model(
```

Comment:
How is the train step getting launched? Isn't the main invocation needed?

Reply:
Training step is invoked from the pipeline. Removed the debugging code